File: domox-service/src/test/java/domox/nlp/CoreNlpTest.java
import org.junit.jupiter.api.Test;
import org.testcontainers.containers.GenericContainer;
import org.testcontainers.junit.jupiter.Container;
import org.testcontainers.junit.jupiter.Testcontainers;

@Testcontainers
public class CoreNlpTest {

    @Container
    private static final GenericContainer<?> coreNlp = new GenericContainer<>("corenlp:latest")
            .withExposedPorts(9000)
            .withEnv("JVM_OPTS", "-Xmx4g");

    @Test
    void testCoreNlp() {
        // Your test logic here, e.g., connect to coreNlp.getMappedPort(9000)
        String host = coreNlp.getHost();
        Integer port = coreNlp.getMappedPort(9000);
        // Use host and port to interact with CoreNLP
    }
}

File: domox-service/src/test/kotlin/domox/nlp/CoreNlpTest.kt
package domox.nlp

import org.junit.jupiter.api.Disabled
import org.testcontainers.containers.GenericContainer
import org.testcontainers.utility.DockerImageName
import kotlin.test.Test

@Disabled("Requires Docker environment with CoreNLP container")
class CoreNLPTest {
    companion object {
        private val CORE_NLP_IMAGE = DockerImageName.parse("graham3333/corenlp-complete")
    }

    @Test
    fun `test CoreNLP container`() {
        GenericContainer(CORE_NLP_IMAGE).apply {
            withExposedPorts(9000)
            start()

            val coreNlpUrl = "http://${host}:${getMappedPort(9000)}/"
            println("CoreNLP running at: $coreNlpUrl")

            stop()
        }
    }
}
